msgpack/_unpacker.pyx

```diff
@@ -145,6 +147,9 @@ def unpackb(object packed, object object_hook=None, object list_hook=None,
         else:
             raise
     else:
+        warnings.warn("unpacking %s requires old buffer protocol,"
```

Comment:
Should this warning be issued after the attempt to get the buffer? If the object does not support any protocol, it will still result in a ``TypeError``, but the message may be a bit confusing.